test/plugin.js

```diff
@@ -61,7 +61,7 @@ describe('Plugin tests', function() {
     var plugin;
 
     beforeEach(function() {
-      plugin = new Plugin(new Session(new Connection, 'id'), 'name', 'id');
+      plugin = new Plugin(new Session(new Connection('id', {address: 'address'}), 'id'), 'name', 'id');
```

Comment:
I did not pay too much attention when reviewing specs. But you should not need to adjust session/plugin specs when changing constructor of Connection. You should use mocks for Session/Connection.